### ISSUE TITLE
[Bug] g.gui.iclass failed to load existing vector training data.

### ISSUE THREAD

**Issue by tgrippa** [OSGeo/grass#1019]
**Describe the bug**
Module g.gui.iclass failed to load existing vector training data.

**To Reproduce**

1. Launch GRASS on nc_spm_08_grass7 mapset
2. Download the zip file provided in this bug report and unzip it.
3. Import the training set (vector): v.unpack input=C:\Users\tais\Downloads\training.pack output=training_lc
4. Define an imagery group: i.group group=bands subgroup=bands input=lsat7_2002_20,lsat7_2002_30,lsat7_2002_40,lsat7_2002_50
5. Launch g.gui.iclass : g.gui.iclass subgroup=bands trainingmap=training_nc
6. See error in the pop-up 

This error also appears when g.gui.iclass is launched without specifying a training vector map and the user try to load the vector map using the GUI tools. 

**Expected behavior**
The module is supposed to be able to load existing training polygons provided as a vector layer.

**System description (please complete the following information):**

- Operating System: Both Windows7 and Ubuntu 18.04
- GRASS GIS version: 7.9.dev (windows) and 7.8.3 (ubuntu)

[training.zip](https://github.com/OSGeo/grass/files/5372556/training.zip)

**Comment by neteler** [comment]
I can confirm this problem on Fedora 32:

```
g.gui.iclass subgroup=bands trainingmap=training_nc
Loading training map <training_nc@user1>...
G__open(read): Unable to open '/home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13019130/frmt': No such file or directory
G__open(read): Unable to open '/home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13019130/frmt': No such file or directory

Segmentation fault (core dumped)
```

To be sure that the dataset isn't broken, I tried:

- display in d.mon (works ok)
- open in digitizer (works ok)
- run g.gui.iclass (fails):
- extract into a new map and run g.gui.iclass (fails, same error):

```
GRASS nc_spm_08_grass7/user1:tmp > v.db.select training_nc
cat|class|color|n_cells|band1_min|band2_min|band3_min|band4_min|band1_mean|band2_mean|band3_mean|band4_mean|band1_max|band2_max|band3_max|band4_max
1|class_1|255:0:0:255|||||||||||||
2|class_2|0:128:0:255|||||||||||||
3|class_3|255:255:128:255|||||||||||||
4|class_4|0:0:255:255|||||||||||||

GRASS nc_spm_08_grass7/user1:tmp > v.extract input=training_nc output=training_nc2
Extracting features...
 100%
Building topology for vector map <training_nc2@user1>...
Registering primitives...
Building areas...
 100%
Attaching islands...
 100%
Attaching centroids...
 100%
Writing attributes...

GRASS nc_spm_08_grass7/user1:tmp > g.gui.iclass subgroup=bands trainingmap=training_nc2
Loading training map <training_nc2@user1>...
G__open(read): Unable to open '/home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13025600/frmt': No such file or directory
G__open(read): Unable to open '/home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13025600/frmt': No such file or directory

Segmentation fault (core dumped)
```

Test with debugging on:

```
g.gisenv set="DEBUG=2"
g.gui.iclass subgroup=bands trainingmap=training_nc
...
D1/2: V1_open_old_nat(): name = training_nc mapset = user1
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/vector/training_nc/coor
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/vector/training_nc/coor
D2/2: 	file open: read (mode = r)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/vector/training_nc/coor
D1/2: get coor info: /home/mneteler/grassdata/nc_spm_08_grass7/user1/vector/training_nc/coor
D1/2: Vect_coor_info(): Info->size = 3010, Info->mtime = 1602603524
D2/2: dig__read_head(): name = 'training_nc'
D2/2: Coor header: file version 5.1 , supported from GRASS version 5.1
D2/2:   byte order 0
D2/2:   header size 18
D2/2:   with_z 0
D2/2:   coor size 3010
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/vector/training_nc/coor
D1/2: get coor info: /home/mneteler/grassdata/nc_spm_08_grass7/user1/vector/training_nc/coor
D1/2: Vect_coor_info(): Info->size = 3010, Info->mtime = 1602603524
D1/2: coor size in head = 3010, real coor file size= 3010
D2/2: dig_file_load ()
D2/2:   requested mode = 2
D2/2:   size = 3010
D2/2:   file was not loaded to the memory
D1/2: Vect__open_old(): vector opened on level 2
D1/2: Vect_read_dblinks(): map = training_nc, mapset = user1
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/vector/training_nc/dbln
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/vector/training_nc/dbln
D2/2: 	file open: read (mode = r)
D1/2: dbln: 1/training_nc|training_nc|cat|$GISDBASE/$LOCATION_NAME/$MAPSET/sqlite/sqlite.db|sqlite
D1/2: field = 1 name = training_nc, table = training_nc, key = cat, database = $GISDBASE/$LOCATION_NAME/$MAPSET/sqlite/sqlite.db, driver = sqlite
D1/2: Dblinks read
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/vector/training_nc/hist
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/vector/training_nc/hist
D2/2: 	file open: read (mode = r)
D1/2: Vect_Rewind(): name = training_nc level = 2
D1/2: Vect_get_dblink(): link = 0
D1/2: grass.script.core.start_command(): g.tempfile -d pid=1303663
D1/2: G_set_program_name(): g.tempfile
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=0)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/1303663.0
D2/2: G_tempfile_pid(): /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/1303663.0
D1/2: Vect_open_tmp_new(): name = 'trAreas13036630' with_z = 0
D1/2: Vect_open_new(): name = trAreas13036630 with_z = 0 is_tmp = 2
D1/2: Vect_set_thresh(): thresh = 0.000000
D2/2:  using non-direct format
D1/2: Vect__open_old(): name = trAreas13036630, mapset = user1, layer = NULL, update = 0, head_only = 1, is_tmp = 2
D1/2: Vect_set_thresh(): thresh = 0.000000
D1/2: dig_spidx_init(), 2 dims
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=1)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=1)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630/head
D1/2: Map: name = trAreas13036630, mapset = user1, temporary = 2
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630/frmt
G__open(read): Unable to open '/home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630/frmt': No such file or directory
D1/2: Vector format: 0 (native)
D1/2: Vect_set_thresh(): thresh = 0.000000
D1/2: Vect__read_head(): vector = trAreas13036630@user1
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=1)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630/head
D2/2: 	file open: read (mode = r)
D1/2: Vect_set_thresh(): thresh = 0.000000
D1/2: Level request = 1
D1/2: Vect__open_old(): vector opened on level 1
D1/2: Vect_read_dblinks(): map = trAreas13036630, mapset = user1
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=1)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630/dbln
D2/2: 	file open: read (mode = r)
D1/2: Dblinks read
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630/hist
D2/2: 	file open: read (mode = r)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=1)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630/dbln
D1/2: dbln file: /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630/dbln
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=1)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630
D1/2: Vect_close(): name = trAreas13036630, mapset = user1, format = 0, level = 1, is_tmp = 2
D1/2: close history file
...
D1/2: dig__write_head()
D1/2: write coor size (3010) to head
D2/2: coor body offset 18
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=1)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630/head
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: 	file open: new (mode = w)
D1/2: Vect_write_dblinks(): map = trAreas13036630, mapset = user1
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=1)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630/dbln
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: 	file open: new (mode = w)
D1/2: Dblinks written
D1/2: V1_close_nat(): temporary map <trAreas13036630> TO BE DELETED
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=1)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630

D1/2: Vect_close(): name = training_nc, mapset = user1, format = 0, level = 2, is_tmp = 0
D1/2: spatial index file closed
D1/2: close history file
D1/2: V1_close_nat(): name = training_nc mapset= user1
D1/2: grass.script.core.start_command(): g.gisenv -n
D1/2: G_set_program_name(): g.gisenv
D1/2: Vect__open_old(): name = trAreas13036630, mapset = user1, layer = NULL, update = 1, head_only = 0, is_tmp = 1
D1/2: Vect_set_thresh(): thresh = 0.000000
D1/2: dig_spidx_init(), 2 dims
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=1)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1
D2/2: G__temp_element(): .tmp/oboe (tmp=1)
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/.tmp/oboe/vector/trAreas13036630/head
D2/2: G_file_name(): path = /home/mneteler/grassdata/nc_spm_08_grass7/user1/vector/trAreas13036630/head
Segmentation fault (core dumped)
D1/2: G_set_program_name(): g.gisenv
D1/2: G_set_program_name(): g.gisenv
```

It seems to be a `g.gui.iclass` specific problem.

**Comment by neteler** [comment]
Fixed in #1020 and backported to 7.8.5.
